lib/Runtime/Base/FunctionBody.h

```diff
@@ -316,13 +316,28 @@ namespace Js
     public:
         // These are public because we don't manage them nor their consistency;
         // the user of this class does.
-        void * address;
+        union address_type_ {
+            address_type_(void *ptr):asPtr(ptr) {}
+            address_type_(Js::JavascriptMethod method):asJsMethod(method) {}
 
-        ProxyEntryPointInfo(void* address, ThreadContext* context = nullptr):
+            void *asPtr;
```

Comment:
At the page https://isocpp.org/wiki/faq/cpp11-language-types#generalized-unions there is a sentence: `it’s illegal to write one member and then read another`.
So in my point of view, it is incorrect to use any of the `address_type_` union constructor and then execute a code like 

```
Assert(entryPoint->address.asPtr != nullptr);
return entryPoint->address.asJsMethod;
```
because it will read a field that wasn't directly initialised. Am I wrong?